framework/src/meshgenerators/OverlayMeshGenerator.C

```diff
@@ -0,0 +1,138 @@
+  params.addParam<processor_id_type>(
+      "num_cores_for_partition",
+      0,
+      "Number of cores for partitioning the graph (dafaults to the number of MPI ranks)");
+  params.addRangeCheckedParam<unsigned>(
+      "num_side_layers",
+      2,
+      "num_side_layers>=1 & num_side_layers<5",
+      "Number of layers of off-processor side neighbors is reserved during mesh generation");
+  MooseEnum partition("graph linear square", "graph", false);
+  params.addParam<MooseEnum>(
+      "partition", partition, "Which method (graph linear square) use to partition mesh");
+  MooseEnum elem_types = MooseMesh::elemTypes();
+  params.addParam<MooseEnum>("elem_type",
+                             elem_types,
+                             "The type of element from libMesh to "
+                             "generate (default: linear element for "
+                             "requested dimension)");
+
+  params.addRangeCheckedParam<Real>(
+      "bias_x",
+      1.,
+      "bias_x>=0.5 & bias_x<=2",
+      "The amount by which to grow (or shrink) the cells in the x-direction.");
+  params.addRangeCheckedParam<Real>(
+      "bias_y",
+      1.,
+      "bias_y>=0.5 & bias_y<=2",
+      "The amount by which to grow (or shrink) the cells in the y-direction.");
+  params.addRangeCheckedParam<Real>(
+      "bias_z",
+      1.,
+      "bias_z>=0.5 & bias_z<=2",
+      "The amount by which to grow (or shrink) the cells in the z-direction.");
+  return params;
+}
+
+OverlayMeshGenerator::OverlayMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _dim(getParam<MooseEnum>("dim")),
+    _mesh_name(getParam<MeshGeneratorName>("input"))
+{
+  // Declare that all of the meshes in the "inputs" parameter are to be used by
+  // a sub mesh generator
+  declareMeshForSub("input");
+
+  _input_mesh = &getMeshByName(_mesh_name);
+
+  auto params = _app.getFactory().getValidParams("DistributedRectilinearMeshGenerator");
+
+  params.set<MooseEnum>("dim") = _dim;
+
+  params.set<dof_id_type>("nx") = getParam<dof_id_type>("nx");
+  params.set<dof_id_type>("ny") = getParam<dof_id_type>("ny");
+  params.set<dof_id_type>("nz") = getParam<dof_id_type>("nz");
+
+  params.set<Real>("xmin") = 0;
+  params.set<Real>("ymin") = 0;
+  params.set<Real>("zmin") = 0;
+  params.set<Real>("xmax") = 1;
+  params.set<Real>("ymax") = 1;
+  params.set<Real>("zmax") = 1;
+
+  params.set<Real>("bias_x") = getParam<Real>("bias_x");
+  params.set<Real>("bias_y") = getParam<Real>("bias_y");
+  params.set<Real>("bias_z") = getParam<Real>("bias_z");
+
+  params.set<unsigned>("num_side_layers") = getParam<unsigned>("num_side_layers");
+  params.set<processor_id_type>("num_cores_for_partition") =
+      getParam<processor_id_type>("num_cores_for_partition");
+
+  params.set<MooseEnum>("partition") = getParam<MooseEnum>("partition");
+  params.set<MooseEnum>("elem_type") = getParam<MooseEnum>("elem_type");
```

Comment:
you can replace a lot of this with just params.applyParams

or applySelectedParams if the overlap isnt neat